### ISSUE TITLE
Losing reference to $CAGEBREAK_SOCKET

### ISSUE THREAD

**Issue by kinleyd** [project-repo/cagebreak#17]
After starting Cagebreak, initial configuration calls to the IPC socket work as usual.
However, after cagebreak/config is executed, if I run this in a terminal:
$ rlwrap nc -U $CAGEBREAK_SOCKET                            
I get this message:
Ncat: You have to specify path to a socket to connect to. QUITTING.
At this point 'echo $CAGEBREAK_SOCKET' returns nothing.

If I save the value of $CAGEBREAK_SOCKET in cagebreak/config (in the call to my initialization shell script, cb-netcat):
'echo $CAGEBREAK_SOCKET > socket-path'
and use that socket path to access the IPC socket, it works. 

So the ipc socket itself isn't the problem just the $CAGEBREAK_SOCKET seems to be getting lost at some point.

**Comment by project-repo** [comment]
Interesting...

I cannot reproduce your issue. For me, `$CAGEBREAK_SOCKET` is set after
the start of cagebreak.

Could you please describe the issue more precisely?

  * Which cagebreak version are you using?
  * Is the shell you use to interact with the socket special in some way?
    * Is it in a different TTY?
    * Is it started before or after executing cagebreak?
    * Is your shell bash or sh or something else?
    * What happens if you run `nc` directly without `rlwrap`?
  * Can you determine more precisely when (point in time relative to
    execution of cagebreak) `$CAGEBREAK_SOCKET` is lost?
  * Is there any other detail you would like to share? (You mentioned
    emacs in another issue, is there any special way to start programs
    from emacs which somehow messes with env vars? Or any other programs
    you know that might have any influence?)

cheers
project-repo

**Comment by kinleyd** [comment]
SCAGEBREAK_SOCKET value was accessible at all points in cagebreak/config and the shell script called for initial configuration.

Anyway, I managed to identify the problem: Initially I was starting the Emacs daemon in config, but later changed to starting it as a systemd service, emacs.service. Set this way Emacs starts before Cagebreak and doesn't pick up $CAGEBREAK_SOCKET. The terminal I start netcat in is an Emacs package, vterm, which of course doesn't see $CAGEBREAK_SOCKET. I thought I had tested running netcat in alacritty (non-Emacs terminal) but apparently I hadn't, because when I tried it just now, it works while vterm doesn't. 

So if I want to continue using systemd and emacs.service, I just have to continue to add a bit of setenv settings in Emacs to pick up $CAGEBREAK_SOCKET.

Dammit, sorry to have bothered you!

**Comment by kinleyd** [comment]
$CAGEBREAK_SOCKET value was accessible at all points in cagebreak/config and the shell script called for initial configuration.

Anyway, I managed to identify the problem: Initially I was starting the Emacs daemon in config, but later changed to starting it as a systemd service, emacs.service. Set this way Emacs starts before Cagebreak and doesn't pick up $CAGEBREAK_SOCKET. The terminal I start netcat in is an Emacs package, vterm, which of course doesn't see $CAGEBREAK_SOCKET. I thought I had tested running netcat in alacritty (non-Emacs terminal) but apparently I hadn't, because when I tried it just now, it works while vterm doesn't. 

So if I want to continue using systemd and emacs.service, I just have to continue to add a bit of setenv settings in Emacs to pick up $CAGEBREAK_SOCKET.

Dammit, sorry to have bothered you!
